types/module/configurator.go

```diff
@@ -45,6 +52,31 @@ type configurator struct {
 
 	// migrations is a map of moduleName -> fromVersion -> migration script handler
 	migrations map[string]map[uint64]MigrationHandler
+
+	registryCache *protoregistry.Files
+}
+
+// RegisterService implements the grpc.Server interface
+func (c configurator) RegisterService(sd *googlegrpc.ServiceDesc, ss interface{}) {
+	if c.registryCache == nil {
+		var err error
+		c.registryCache, err = proto.MergedRegistry()
+		if err != nil {
+			panic(err)
```

Comment:
Let's not panic, but instead have error field on `configurator` and then error later